model/plugin_kvset_options.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package model
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+type PluginKVSetOptions struct {
+	EncodeJSON      bool
+	Atomic          bool
+	OldValue        interface{}
+	ExpireInSeconds int64
+}
+
+func (opt *PluginKVSetOptions) IsValid() *AppError {
+	if !opt.Atomic && opt.OldValue != nil {
+		return NewAppError("PluginKVSetOptions.IsValid", "model.plugin_kvset_options.is_valid.old_value.app_error", map[string]interface{}{}, "OldValue should not be defined on non atomic sets", http.StatusBadRequest)
+	}
+
+	return nil
+}
+
+func (opt *PluginKVSetOptions) GetPluginKeyValue(pluginId string, key string, value interface{}) (*PluginKeyValue, *AppError) {
```

Comment:
feels really odd to have `PluginKVSetOptions` depend on `PluginKeyValue` by defining a function that returns `PluginKeyValue`. Should this just be a helper instead?

Reply:
I don't really see why this is odd. Can you please elaborate?